platform/android/dir_access_jandroid.cpp

```diff
@@ -154,7 +154,7 @@ Error DirAccessJAndroid::change_dir(String p_dir){
 
 String DirAccessJAndroid::get_current_dir(){
 
-	return "/"+current_dir;
+	return "res://"+current_dir;
```

Comment:
https://github.com/godotengine/godot/blob/master/platform/android/os_android.cpp#L77-L115

If I understand this code right, `dir_access_jandroid.cpp` handles only "res://" on running on Android device.
The problem is that `res://path` is changed to `/path`.
So, it will use `DirAccess::ACCESS_FILESYSTEM` after `var path = dir.get_current_dir().plus_file("sub_dir_name")`

So, I don't think it makes other trouble.
And I tested it with exported apk, run on Android device.